examples/all-clusters-app/all-clusters-common/all-clusters-app.matter

```diff
@@ -3992,70 +4036,123 @@ cluster Thermostat = 513 {
     kSleep = 9;
   }
 
-  bitmap DayOfWeek : bitmap8 {
-    kSunday = 0x1;
-    kMonday = 0x2;
-    kTuesday = 0x4;
-    kWednesday = 0x8;
-    kThursday = 0x10;
-    kFriday = 0x20;
-    kSaturday = 0x40;
-    kAway = 0x80;
+  enum TemperatureSetpointHoldEnum : enum8 {
+    kSetpointHoldOff = 0;
+    kSetpointHoldOn = 1;
+  }
+
+  enum ThermostatRunningModeEnum : enum8 {
+    kOff = 0;
+    kCool = 3;
+    kHeat = 4;
+  }
```

Comment:
That's interesting, yeah it's not in there in the master spec.  Thermostat Running Mode is using SystemModeEnum in the spec, the tool still generated this though (the tool has some built in exceptions where neccesary to keep compatibility).

The XML in master also has this enum and while it's not directly referenced in the Thermostat Running Mode its an enum8, with the same values.  I gotta wonder if this was added to the XML but not changed in the spec?  Since the rest of the values in System Mode don't really make a long of sense for Running mode (which is used to indicate if Auto is heating or cooling at the moment).

Maybe a spec change is in order here?  I'll hunt down the git history of this enum.

Reply:
Looks like this is where it was added.

https://github.com/project-chip/connectedhomeip/commit/f2d408d26acfce2e050704ca6626bf5028072f8b

We are going to have to resolve this, what's the history on the spec?  Did the spec ever use ThermostatRunningMode?

Reply:
No, I added it to #7848 so that it would be in sync with the SDK.